stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -319,6 +319,7 @@ public final class _DataStorage {
     
     @inline(__always)
     public func append(_ bytes: UnsafeRawPointer, length: Int) {
+        precondition(length >= 0, "Length of appending bytes must be positive")
```

Comment:
isn't this 'non-negative' if we are going to allow zero.  also,  i was going to make a joke about grumpy numbers not being welcome. :)